src/script/main/app.ts

```diff
@@ -401,23 +400,21 @@ class App {
       }
 
       const selfUser = await this.initiateSelfUser();
-      if (this.apiClient.backendFeatures.isFederated) {
+
+      if (this.apiClient.backendFeatures.isFederated && this.repository.storage.storageService.db) {
         // Migrate all existing session to fully qualified ids (if need be)
-        const sessions = this.repository.storage.storageService.db?.sessions;
-        const domain = this.apiClient.context?.domain;
-        if (sessions && domain) {
-          await migrateToQualifiedSessionIds(sessions, domain);
-        }
+        await migrateToQualifiedSessionIds(this.repository.storage.storageService.db.sessions, context.domain ?? '');
```

Comment:
The condition on `domain` is important here. We should never migrate with an empty domain!!!